### PR TITLE
Render application icon

### PR DESCRIPTION
**Pull request by shankar2105**
Fetch app icon from application releases and render on UI.

### CODE REVIEW

File: app/actions/alias/app_manager_actions.ts [hunk 1]
```diff
@@ -51,6 +54,28 @@ export const TYPES = {
     ALIAS_SKIP_APP_UPDATE: 'ALIAS_SKIP_APP_UPDATE'
 };
 
+const getAppIcon = ( application ): Promise<string> => {
```
Comment (joshuef) [inline app/actions/alias/app_manager_actions.ts]: I think we should name this differently to indicate that this is getting the icon from a remote server. Perhaps `fetchAppIconFromServer`, as we may also want to fetch icons from cache too

File: app/actions/alias/app_manager_actions.ts [hunk 2]
```diff
@@ -51,6 +54,28 @@ export const TYPES = {
...
+    return new Promise( async ( resolve ) => {
+        try {
+            const filename = `${application.id}.png`;
+            const filePath = path.resolve( os.tmpdir(), filename );
```
Comment (joshuef) [inline app/actions/alias/app_manager_actions.ts]: I think we should probably do a few things when fetching:

1. Only fetch if there's a newer version of the app available (than what we know). (If not, we pull from local cache if available)
2. Store this icon locally to not have to fetch every time.

File: app/actions/alias/app_manager_actions.ts [hunk 3]
```diff
@@ -60,10 +85,12 @@ const fetchAppListFromServer = async (): Promise<void> => {
...
         const response = await request( APPLICATION_LIST_SOURCE );
         const apps = JSON.parse( response );
-        logger.debug( 'Application list retrieved sucessfully' );
+        logger.debug( 'Application list retrieved successfully' );
 
-        Object.keys( apps.applications ).forEach( ( theApp ) => {
-            store.dispatch( updateAppInfoIfNewer( apps.applications[theApp] ) );
+        Object.keys( apps.applications ).forEach( async ( theAppId ) => {
+            const theApp = apps.applications[theAppId];
+            theApp.iconUrl = await getAppIcon( theApp );
```
Comment (joshuef) [inline app/actions/alias/app_manager_actions.ts]: this should probably do something similar w/ `updateAppIconIfNewer`)

File: app/definitions/application.d.ts [hunk 4]
```diff
@@ -61,6 +61,7 @@ export interface App {
     packageName: string;
     repositoryOwner: string;
     repositorySlug: string;
+    iconUrl?: string;
```
Comment (joshuef) [inline app/definitions/application.d.ts]: I think `iconPath` is probably more correct here, as we're fetching from the local system in the end...

File: app/reducers/initialAppManager.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ export const initialAppManager: AppManagerState = {
             packageName: 'safe-browser',
             repositoryOwner: 'maidsafe',
             repositorySlug: 'safe_browser',
```
Comment (joshuef) [inline app/reducers/initialAppManager.ts]: I think we should ship icons for all apps we have in the default list with the application. That way there's always a backup icon. And only if there's something newer would we update it (and therefore store it locally)

Comment (shankar2105) [reply app/reducers/initialAppManager.ts]: Before that should we need to decide on storing and updating the app list locally?

Comment (joshuef) [reply app/reducers/initialAppManager.ts]: I think we should also do this yeh. One only override with a newer released list from server

Comment (shankar2105) [reply app/reducers/initialAppManager.ts]: @joshuef I've updated the code to use a locally shipped icon if there is any issue in fetching the icon from server.

**Pull request merged**